truffle/com.oracle.truffle.api.instrumentation/src/com/oracle/truffle/api/instrumentation/TruffleInstrument.java

```diff
@@ -0,0 +1,233 @@
+            this.in = in;
+            this.err = err;
+            this.out = out;
+        }
+
+        /**
+         * Input associated with {@link com.oracle.truffle.api.vm.PolyglotEngine} this
+         * {@link TruffleInstrument instrument} is being executed in.
+         *
+         * @return reader, never <code>null</code>
+         */
+        public InputStream in() {
+            return in;
+        }
+
+        /**
+         * Standard output writer for {@link com.oracle.truffle.api.vm.PolyglotEngine} this
+         * {@link TruffleInstrument instrument} is being executed in.
+         *
+         * @return writer, never <code>null</code>
+         */
+        public OutputStream out() {
+            return out;
+        }
+
+        /**
+         * Standard error writer for {@link com.oracle.truffle.api.vm.PolyglotEngine} this
+         * {@link TruffleInstrument instrument} is being executed in.
+         *
+         * @return writer, never <code>null</code>
+         */
+        public OutputStream err() {
+            return err;
+        }
+
+        /**
+         * Evaluates source of (potentially different) language using the current context.The names
+         * of arguments are parameters for the resulting {#link CallTarget} that allow the
+         * <code>source</code> to reference the actual parameters passed to
+         * {@link CallTarget#call(java.lang.Object...)}.
+         *
+         * @param source the source to evaluate
+         * @param argumentNames the names of {@link CallTarget#call(java.lang.Object...)} arguments
+         *            that can be referenced from the source
+         * @return the call target representing the parsed result
+         * @throws IOException if the parsing or evaluation fails for some reason
+         */
+        @SuppressWarnings("static-method")
+        public CallTarget parse(Source source, String... argumentNames) throws IOException {
+            return InstrumentationHandler.ACCESSOR.parse(null, source, null, argumentNames);
+        }
+
+    }
+
+    /**
+     * Annotation to register {@link TruffleInstrument instrumentation} instances for automatic
+     * discovery.
+     */
+    @Retention(RetentionPolicy.SOURCE)
+    @Target(ElementType.TYPE)
+    public @interface Registration {
+
+        /**
+         * A custom machine identifier for this instrumentation. If not defined then the fully
+         * qualified class name is used.
+         */
+        String id() default "";
+
+        /**
+         * The name of the instrumentation in an arbitrary format for humans.
+         */
+        String name() default "";
+
+        /**
+         * The version for instrumentation in an arbitrary format.
+         */
+        String version() default "";
+
+        /**
+         * Whether or not the annotated instrumentation should be enabled by default.
+         */
+        boolean autostart() default false;
```

Comment:
Could the docs please link to the mechanism how I can disable these things?

And why is it a good idea to enable anything be default? How can I disable this feature? I don't want any 'surprise' features in my language just because I import some package.